examples/widgets/span_selector.py

```diff
@@ -32,10 +32,12 @@ def onselect(xmin, xmax):
 
     region_x = x[indmin:indmax]
     region_y = y[indmin:indmax]
-    line2.set_data(region_x, region_y)
-    ax2.set_xlim(region_x[0], region_x[-1])
-    ax2.set_ylim(region_y.min(), region_y.max())
-    fig.canvas.draw()
+
+    if len(region_x) >= 2:
+        line2.set_data(region_x, region_y)
+        ax2.set_xlim(region_x[0], region_x[-1])
+        ax2.set_ylim(region_y.min(), region_y.max())
+        fig.canvas.draw()
```

Comment:
I'd rather use 
```
if xmin == xmax:
    return
```
at the top of the function, which is a bit clearer on the semantics.

Reply:
This is not true, because there is possibility of empty list which will give error in this case.

Reply:
You're right.